applications/asset_tracker_v2/src/modules/util_module.c

```diff
@@ -305,9 +305,10 @@ static void on_all_states(struct util_msg_data *msg)
 	 * that the application is ready to shutdown. This ensures a graceful
 	 * shutdown.
 	 */
-	if (reboot_ack_cnt >= module_active_count_get()) {
-		k_delayed_work_submit(&reboot_work,
-				      K_SECONDS(50));
+	if (reboot_ack_cnt >= module_active_count_get() - 1) {
```

Comment:
Instead of hardcoding exactly `1` module to be allow to be skipped and deriving the "needs to ACK reboot" from [starting the module](https://github.com/nrfconnect/sdk-nrf/blob/27c71b2f070a8e3cca0cde47ec08e162ae532aa2/applications/asset_tracker_v2/src/modules/modules_common.c#L73) add a property to modules that tell whether they implement shutdown ack or not, and maintain this as `wait_for_reboot_ack_count`.

Reply:
This PR is a bug fix we need to get in before 1.5. Your suggestion would involve additional changes that I'm afraid we don't have time to incorporate. Nevertheless, this is something we can have a look at. Thank you for the suggestion.